### DISCUSSION TITLE
Docs Bug, localhost != 0.0.0.0

### DISCUSSION THREAD

**Discussion by mattfbacon** [tokio-rs/axum#1982] [Q&A]
The hello world example:

```rust
use axum::{
    routing::get,
    Router,
};

#[tokio::main]
async fn main() {
    // build our application with a single route
    let app = Router::new().route("/", get(|| async { "Hello, World!" }));

    // run it with hyper on localhost:3000
    axum::Server::bind(&"0.0.0.0:3000".parse().unwrap())
        .serve(app.into_make_service())
        .await
        .unwrap();
}
```

says "on localhost:3000" but binds to 0.0.0.0 which is not localhost.

**Comment by davidpdrsn** [comment]
What's the bug? It's still accessible on `localhost:3000`.

**Reply by jplatte** [reply]
I guess "run it in localhost" is just confusing, if you were to do that (via actual localhost IP) it would *only* be available locally.

**Reply by mattfbacon** [reply]
I find it misleading. localhost resolves to 127.0.0.1 but  that is not used here.

**Reply by mattfbacon** [reply]
Anyway, is there a reason you're *opposed* to changing it?

**Reply by davidpdrsn** [reply]
> Anyway, is there a reason you're opposed to changing it?

No reason. We can change it. Sounds very important to you so feel free to make a PR.

**Reply by mattfbacon** [reply]
#1984

**Reply by traitman** [reply]
it's original is not like that.

orig is bind to `127.0.0.1:3000`, and I remember someone send a PR change it to `0.0.0.0:3000`

**Comment by jblachly** [comment]
0.0.0.0 is all interfaces, which is a security risk if someone believes this address is localhost 

Edit: meant this in reply to @davidpdrsn question
